### PR TITLE
command: Should not consume 'print' to output default remote info

### PR DESCRIPTION
**Pull request by eggqq007**
1. should not consume 'print' to output default remote info,
   otherwise the --quiet would not work.

* [ ] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [ ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/command/remote.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def run(self):
         if self.args.name is None and not self.args.unset:
             conf = self.config.read(self.args.level)
             try:
-                print(conf["core"]["remote"])
+                ui.write(conf["core"]["remote"])
```
Comment (skshetry) [inline dvc/command/remote.py:89]: @eggqq007, could you please describe your use case? Do you want to check if the default is set or not?

Comment (skshetry) [reply dvc/command/remote.py:89]: We don't treat `--quiet` as a no output mode. We still output the data that the user asked for, but silence all the other outputs that DVC usually throws (so that `dvc metrics diff --show-md` still works with `--quiet`, etc).

This is a bit of an edge-case here as you may want to either check what the default is or test if the default remote has been set.

Comment (eggqq007) [reply dvc/command/remote.py:89]: Thanks for the reply. emm, I think  updating the dvc document in https://dvc.org/doc/command-reference/remote/add  is a candidate.  It says "-q, --quiet - do not write anything to standard output. Exit with 0 if no problems arise, otherwise 1."

Comment (dberenbaum) [reply dvc/command/remote.py:89]: Current behavior seems consistent with the guideline in https://github.com/iterative/dvc/discussions/5891. I think it even gives the behavior you are looking for @eggqq007, but it's inverted (there is output printed only if the default remote has been set). The command reference definitely needs an update, though, and probably for other commands also.

Comment (skshetry) [reply dvc/command/remote.py:89]: Yeah, I think this could be interpreted as both, so we should probably not output on quiet here, as it's may not just be the data user asked for. And, as per guideline, we only print the output on `--quiet` where it makes the most sense.

**Review by skshetry — APPROVED** [review]

File: dvc/command/remote.py (hunk 1 above)
Comment (dberenbaum) [reply dvc/command/remote.py:89]: Sorry, that's not what I meant, @skshetry. I was suggesting that we close this PR and make a docs update.

Currently, dvc prints output on `--quiet` if the default remote has been set but not if it hasn't been set. As you mention, the default remote name may be what the user is requesting. Even if the user only wants to know whether the default remote has been set, it seems useful to print something only when the remote has been set, so that users could check not only by exit code but also by whether there was any output. By never printing the output on `--quiet`, isn't it making both use cases more difficult?

Comment (skshetry) [reply dvc/command/remote.py:89]: Ahh, okay. I was mostly sticking to the first point of the guideline:

> If the command's (or, the option) that sole purpose is to display the information, let show it in any case.

I guess we need to clarify if this belongs to the second one or not:

> If the information is very important to the user, print warnings and error messages to the stderr even in --quiet mode wherever appropriate.

I have mostly chosen to err on the side of caution, and support `--quiet` only if the command's/flag's sole purpose is to show that information, if `--quiet` has other uses too, I have kept that so far (as we have discussed, `--quiet` on the `dvc metrics diff` probably is an error on the user's side).

`dvc status` does the same thing when used with `--quiet`, in fact it's what we have been recommending for use in the scripts. 

To resolve this, I guess we need to either suggest alternatives here, or clarify the guideline then.

Comment (dberenbaum) [reply dvc/command/remote.py:89]: I'm not following. Isn't the sole purpose of `dvc remote default` (without any args/options) and `dvc status` to display information? What do you think `dvc remote default --quiet` should show according to the guideline?

Comment (efiop) [reply dvc/command/remote.py:89]: > Isn't the sole purpose of dvc remote default (without any args/options) and dvc status to display information?

Actually no, `dvc status -q` without output is used to indicate whether there are any changes, same as `git status -q`. `dvc remote default` can be used similarly in scripts - to tell user if it has default remote or not. So I think that making `-q` suppress it is justifiable beacuse `dvc remote default` has two purposes: with and without printing.

**Comment by jorgeorpinel** [comment]
Will someone be able to address https://github.com/iterative/dvc.org/issues/2840 too? 🙂

File: dvc/command/remote.py (hunk 1 above)
Comment (dberenbaum) [reply dvc/command/remote.py:89]: Discussed this as a group. I was wrong about `dvc status -q`, which never prints anything. In that case, it seems fine to do the same here and merge this PR. I don't think it can cause much harm, and it aligns with the current documentation. Sorry for the oversight and all the noise here.

**Review by dberenbaum — APPROVED** [review]

**Comment by skshetry** [comment]
Thanks @eggqq007 🙂
